pre-commit

```diff
@@ -6,120 +6,165 @@
 import os
 import sys
 import re
+from subprocess import Popen, PIPE
 from unittest import TestCase, TestSuite, TestLoader, TextTestRunner, skip
 from io import StringIO
 from configparser import ConfigParser
+from functools import lru_cache
+
+GIT_BRANCH = 'HEAD'
+
+def readutf8(st):
+    return st.read().decode('utf-8').strip()
+
+def reads(st):
+    return readutf8(st).strip()
+
+def git_write_tree_index():
+    global GIT_BRANCH
+    cmd = ['git', 'write-tree']
+    with Popen(cmd, stdout=PIPE) as p:
+        GIT_BRANCH = reads(p.stdout)
+
+git_write_tree_index()
+
+def git_open(filename):
+    cmd = ['git', 'cat-file', 'blob', f'{GIT_BRANCH}:{filename}']
+    with Popen(cmd, stdout=PIPE) as p:
+        return StringIO(readutf8(p.stdout))
```

Comment:
Why not `return io.TextIOWrapper(p.stdout, encoding='utf-8')`?

Reply:
Because `p.stdout` will have been closed when we read from this later on